### PR TITLE
validate scheduled time is in the future

### PR DESCRIPTION
**Pull request by britneywright**
For scheduled changes we don't want users setting a time in the past.

Changes in this pull request:
- if scheduled changes are enabled check that the scheduled time is after the current time

I went the current time, but we could easily change it to an hour from the current time or some other set time from the current time.

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeValidations.scala [hunk 1]
```diff
@@ -107,6 +107,12 @@ class BatchChangeValidations(
         else NotAMemberOfOwnerGroup(groupId, authPrincipal.signedInUser.userName).invalidNel
     }
 
+//  def validateScheduledTime(scheduledTime: Option[DateTime]): SingleValidation[Unit] =
```
Comment (mitruly) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeValidations.scala]: Commented code should be removed.

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeValidations.scala [hunk 2]
```diff
@@ -507,6 +513,11 @@ class BatchChangeValidations(
   def validateScheduledChange(
       input: BatchChangeInput,
       scheduledChangesEnabled: Boolean): Either[BatchChangeErrorResponse, Unit] =
-    if (scheduledChangesEnabled || input.scheduledTime.isEmpty) Right(())
-    else Left(ScheduledChangesDisabled)
+    (scheduledChangesEnabled, input.scheduledTime.isEmpty) match {
```
Comment (mitruly) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeValidations.scala]: This is more concise:
```
(scheduledChangesEnabled, input.scheduledTime) match {
      case (_, None) => Right(())
      case (true, Some(inputTime)) if inputTime.isAfterNow => Right(())
      case (true, _) => Left(ScheduledTimeMustBeInFuture)
      case (false, _) => Left(ScheduledChangesDisabled)
    }
```

File: modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala [hunk 3]
```diff
@@ -35,6 +35,7 @@ final case class BatchChangeIsEmpty(limit: Int) extends DomainValidationError {
 final case class GroupDoesNotExist(id: String) extends DomainValidationError {
   def message: String = s"""Group with ID "$id" was not found"""
 }
+
```
Comment (mitruly) [inline modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala]: 💥

Comment (britneywright) [reply modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala]: that's valid, not sure when or why but the first few classes don't have spaces between them, the rest do. I'll update the file to have spaces between all of them.

**Comment by mitruly** [comment]
There's one existing functional test that is failing (`test_create_batch_change_with_scheduled_time_and_owner_group_succeeds`) since it is using `datetime.datetime.now()`; it needs to be padded for a future time.

File: modules/api/functional_test/live_tests/batch/create_batch_change_test.py [hunk 4]
```diff
@@ -276,7 +276,7 @@ def test_create_batch_change_with_scheduled_time_and_owner_group_succeeds(shared
     Test successfully creating a batch change with scheduled time and owner group set
     """
     client = shared_zone_test_context.ok_vinyldns_client
-    dt = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
+    dt = (datetime.datetime.now() + datetime.timedelta(1)).strftime('%Y-%m-%dT%H:%M:%SZ')
```
Comment (rebstar6) [inline modules/api/functional_test/live_tests/batch/create_batch_change_test.py]: is this one ms? risky.

Comment (rebstar6) [reply modules/api/functional_test/live_tests/batch/create_batch_change_test.py]: maybe not, can you just specify so its clear? `timedelta(hours=1)` or whatever

File: modules/api/functional_test/live_tests/batch/create_batch_change_test.py [hunk 5]
```diff
@@ -313,6 +313,28 @@ def test_create_scheduled_batch_change_with_zone_discovery_error_without_owner_g
     assert_that(errors, is_("Batch change requires owner group for manual review/scheduled changes."))
 
 
+@pytest.mark.manual_batch_review
+def test_create_scheduled_batch_change_with_scheduled_time_in_the_past_fails(shared_zone_test_context):
+    """
+    Test creating a scheduled batch with a scheduled time in the past
+    """
+    client = shared_zone_test_context.ok_vinyldns_client
+    yesterday = (datetime.datetime.now() - datetime.timedelta(1)).strftime('%Y-%m-%dT%H:%M:%SZ')
```
Comment (rebstar6) [inline modules/api/functional_test/live_tests/batch/create_batch_change_test.py]: same timedelta thing here on all of these

File: modules/api/functional_test/live_tests/batch/create_batch_change_test.py (hunk 4 above)
Comment (britneywright) [reply modules/api/functional_test/live_tests/batch/create_batch_change_test.py]: it's days, that's the default first parameter of timedelta

**Pull request merged**